backends/bmv2/simple_switch/midend.cpp

```diff
@@ -85,6 +86,7 @@ SimpleSwitchMidEnd::SimpleSwitchMidEnd(CompilerOptions& options) : MidEnd(option
         new P4::NestedStructs(&refMap, &typeMap),
         new P4::SimplifySelectList(&refMap, &typeMap),
         new P4::RemoveSelectBooleans(&refMap, &typeMap),
+        new P4::FlattenHeaders(&refMap, &typeMap),
```

Comment:
could you please add this to the p4test midend as well, in a similar place.
then we can see the generated reference outputs in the PR.

Reply:
I'd be happy to.  However, the p4test midend does not include the flattenInterfaceStructs midend pass either.  I think, both passes should be added to p4test.

Reply:
Many tests have failed with 'make check' when adding the two flatten midend passes to p4test.  I am looking into these failures next.  

I checked in code that takes care of rest of the review comments from today.

Reply:
Several reference outputs should change.

Reply:
Yes. I made the changes and checked in code.  I will now look at any pending review comment from today and address it.